contracts/src/L1/rollup/ScrollChain.sol

```diff
@@ -502,14 +754,14 @@ contract ScrollChain is OwnableUpgradeable, PausableUpgradeable, IScrollChain {
         uint256 txHashStartDataPtr;
         assembly {
             txHashStartDataPtr := dataPtr
-            blockPtr := add(chunkPtr, 1) // reset block ptr
+            chunkPtr := add(_chunk, 0x20)
         }
-
         // concatenate tx hashes
-        uint256 l2TxPtr = ChunkCodec.l2TxPtr(chunkPtr, _numBlocks);
+        uint256 l2TxPtr = ChunkCodecV0.l2TxPtr(chunkPtr, _numBlocks);
+        chunkPtr += 1;
```

Comment:
```suggestion
        uint256 blockPtr = chunkPtr + 1;
```

Reply:
I reuse the variable to avoid stack too deep.